src/code/models/codap-connect.ts

```diff
@@ -837,6 +837,35 @@ export class CodapConnect {
     }
   }
 
+  public addNewDataContextAndTable() {
+    const newDataContextName = `NewTable${Math.floor(10000000000 * Math.random())}`;
+    this.codapPhone.call({
+      action: "create",
+      resource: "dataContext",
+      values: {
+        name: newDataContextName,
+        title: tr("~DOCUMENT.CODAP_ACTIONS.TABLES.NEW"),
+        collections: [{
+          name: "Data",
+          attrs: [
+            { "name": tr("~DOCUMENT.CODAP_ACTIONS.TABLES.ATTRIBUTE_NAME") }
+          ]
+        }]
+      }
+    }, (result) => {
+      this.codapPhone.call({
+        action: "create",
+        resource: "component",
+        values: {
+          type: "caseTable",
+          title: tr("~DOCUMENT.CODAP_ACTIONS.TABLES.NEW"),
+          dataContext: newDataContextName,
```

Comment:
There's an assumption here that the call to create the dataContext was successful - could this fail? We never use any of the information returned in the result.